File: 22-Texture-Classification/hw_3_prob_1_a.py
#!/usr/bin/env python2
# -*- coding: utf-8 -*-
"""
Created on Sat Mar 24 16:41:59 2018

@author: abinaya
"""

import numpy as np
import matplotlib.pyplot as plt
plt.close('all')

X = np.loadtxt('energyfeature.txt')
X = X[:,1]
X = X.reshape([12,9])
X = X.astype(int)

label_colors = ['r','g','b','r','b','r','c','c','g','c','b','g']


for i in range(0,9):
    plt.figure()
    plt.bar(X[:,i], [1]*12, color=label_colors)
    plt.title('Feature '+str(i+1))
    plt.xlabel('Feature Values')
    plt.ylabel('Count')
    plt.savefig('feature_'+str(i+1)+'.png')

File: 22-Texture-Classification/hw_3_prob_1_a.m
clc;
clear all;
close all;

filename1 = '/Users/abinaya/USC/Studies/DIP/Homeworks/hw-3/hw_3_answers/hw_3_prob_1_a/hw_3_prob_1_a/energyfeature.txt';
data1 = load(filename1);

energy_1D = data1(:,2);
energy = reshape(energy_1D,[12,9]);
%energy = energy';

for i=1:9
    figure;
    hist(energy(:,i),30);
    title(['Histogram Of Energy Feature ',num2str(i)])
end
